src/components/dialogs/field-hook.js

```diff
@@ -308,56 +323,163 @@ export const useEquipmentTableValues = ({
+    }, [selectedIds, setCreateFilterErr, setIsEdited]);
 
     const handleSetValue = useCallback(
         (index, newValue) => {
             setValues((oldValues) => {
                 let newValues = [...oldValues];
                 newValues[index] = newValue;
-                setIsEdited(true);
                 return newValues;
             });
             setCreateFilterErr('');
+            setIsEdited(true);
         },
         [setCreateFilterErr, setIsEdited]
     );
 
     const handleChangeOrder = useCallback(
-        (index, direction) => {
+        (direction) => {
             const res = [...values];
-            const [item] = res.splice(index, 1);
-            res.splice(index + direction, 0, item);
+            const result = [...selectedIds];
+            result.sort();
+
+            let isContiguous = true;
+            for (let i = 0; i < result.length - 1; i++) {
+                if (result[i + 1] - result[i] !== 1) {
+                    isContiguous = false;
+                    break;
+                }
+            }
+
+            if (isContiguous) {
+                if (direction === -1) {
+                    const [item] = res.splice(result[0] - 1, 1);
+                    res.splice(result[result.length - 1], 0, item);
+                } else {
+                    const [item] = res.splice(result[result.length - 1] + 1, 1);
+                    res.splice(result[0], 0, item);
+                }
+            } else {
+                selectedIds.forEach((elem) => {
+                    const [item] = res.splice(elem, 1);
+                    res.splice(elem + direction, 0, item);
+                });
+            }
+            let array = [...selectedIds];
+            array.sort();
+            for (let i = 0; i < array.length; i++) {
+                array[i] = array[i] + direction;
+            }
+            setSelectedIds(new Set(array));
             setIsEdited(true);
             setValues(res);
         },
-        [values, setIsEdited]
+        [selectedIds, setIsEdited, values]
     );
 
     const commit = useCallback(
         ({ source, destination }) => {
             if (destination === null || source.index === destination.index)
                 return;
             const res = [...values];
+            res.forEach((e) => {
+                e['isChecked'] = false;
+            });
+            selectedIds.forEach((e) => {
+                res[e].isChecked = true;
+            });
+            setSelectedIds(new Set());
+
             const [item] = res.splice(source.index, 1);
             res.splice(
                 destination ? destination.index : values.length,
                 0,
                 item
             );
             setValues(res);
+            let array = [];
+            let j = 0;
+            for (let i = 0; i < res.length; i++) {
+                if (res[i].isChecked) {
+                    array[j] = i;
+                    j++;
+                }
+            }
```

Comment:
Maybe use map to get array from res:
const array = res.filter((val) => val.isChecked)
                .map((val) => {
                return res.indexOf(val);
            })

Reply:
done